internal/authctx/client.go

```diff
@@ -38,26 +38,28 @@ func (cfg *TanzuContext) Setup() (err error) {
 	if err != nil {
 		return err
 	}
+	cfg.TMCConnection.WithHost(cfg.ServerEndpoint)
+	cfg.TMCConnection.Headers.Set("Host", cfg.ServerEndpoint)
 
-	return setup(cfg)
+	return SetAuthCtx(cfg)
 }
 
 // The default transport is needed for mocking. The http mocking library used in testing
 // can only intercept calls if they're made with the default transport.
 func (cfg *TanzuContext) SetupWithDefaultTransportForTesting() (err error) {
 	cfg.TMCConnection = client.NewTestHTTPClientWithDefaultTransport()
-	return setup(cfg)
+	cfg.TMCConnection.WithHost(cfg.ServerEndpoint)
+	cfg.TMCConnection.Headers.Set("Host", cfg.ServerEndpoint)
+
+	return SetAuthCtx(cfg)
 }
 
-func setup(cfg *TanzuContext) (err error) {
+func SetAuthCtx(cfg *TanzuContext) (err error) {
```

Comment:
why is this change needed ?

Reply:
Trying to rename it so that we can reuse it elsewhere